juju/juju
# Revert LXD changes

The following PR reverts the 2.9 changes around LXD. As the LXD project is
using 1.16 version of go and we're using 1.14 there are some issues around
the exact packages we're building with. Running `go mod tidy` on the tip of
2.9 returns an error (see below) which indicates something is using the latest
`io/fs` packages and we're not able to consume them, so it's finding alternative 
dependencies to use from the sum file (at a guess). As stated before, go mod
along with the sum files, are not a lock file, it only describes what's downloaded
and to ensure they're not corrupt.

The fix is simple, revert the LXD changes and work around the LXD status error
by creating an interface to validate against.

```
go mod tidy
github.com/juju/juju/apiserver/logsink imports
        gopkg.in/natefinch/lumberjack.v2 tested by
        gopkg.in/natefinch/lumberjack.v2.test imports
        github.com/BurntSushi/toml imports
        io/fs: package io/fs is not in GOROOT (/usr/local/go/src/io/fs)
github.com/juju/juju/apiserver/logsink imports
        gopkg.in/natefinch/lumberjack.v2 tested by
        gopkg.in/natefinch/lumberjack.v2.test imports
        github.com/BurntSushi/toml tested by
        github.com/BurntSushi/toml.test imports
        testing/fstest: package testing/fstest is not in GOROOT (/usr/local/go/src/testing/fstest)
github.com/juju/juju/apiserver/logsink imports
        gopkg.in/natefinch/lumberjack.v2 tested by
        gopkg.in/natefinch/lumberjack.v2.test imports
        github.com/BurntSushi/toml tested by
        github.com/BurntSushi/toml.test imports
        github.com/BurntSushi/toml/internal/toml-test imports
        embed: package embed is not in GOROOT (/usr/local/go/src/embed)
```

## QA steps


```sh
$ snap install juju --classic
$ /snap/bin/juju bootstrap lxd test
$ juju upgrade-controller --build-agent
$ juju debug-log -m controller
```
Ensure that the single-controller does settle and has started. 

```sh
$ juju ssh -m controller 0
$ juju_engine_report | less
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1943075
